pyproject.toml

```diff
@@ -109,3 +109,101 @@ source-includes = [
 [build-system]
 requires = ["pdm-backend"]
 build-backend = "pdm.backend"
+
+[tool.ruff.lint]
+# see https://docs.astral.sh/ruff/configuration/#using-pyprojecttoml
+# "F" contains autoflake, see https://github.com/astral-sh/ruff/issues/1647
+# add more rules
+select = [
+    # default Ruff checkers as of ruff 0.1.3: E4, E7, E9, F
+    "E4",
+    "E7",
+    "E9",
+    "F",  # pyflakes
+
+    # the rest in alphabetical order:
+    # TODO: "A",   # flake8-builtins
+    # TODO: "ARG", # flake8-unused-arguments
+    "B",   # flake8-bugbear
+    # TODO: "BLE", # flake8-blind-except
+    # TODO: Do I want "COM", # flake8-commas
+    "C4",  # flake8-comprehensions
+    # TODO: "DJ",  # flake8-django
+    # TODO: "DTZ", # flake8-datetimez
+    # TODO: "EM",  # flake8-errmsg
+    "EXE", # flake8-executable
+    # TODO: "FURB", # refurb
+    # TODO: "FBT", # flake8-boolean-trap
+    # TODO: "G",   # flake8-logging-format
+    "I",   # isort
+    "ICN", # flake8-import-conventions
+    "INP", # flake8-no-pep420
+    # TODO: "INT", # flake8-gettext
+    "ISC", # flake8-implicit-str-concat
+    # TODO: "LOG", # flake8-logging
+    "PERF", # perflint
+    "PIE", # flake8-pie
+    "PL",  # pylint
+    "PYI", # flake8-pyi
+    # TODO: "RET", # flake8-return
+    "RSE", # flake8-raise
+    "RUF",
+    # TODO: "SIM", # flake8-simplify
+    "SLF", # flake8-self
+    "SLOT", # flake8-slots
+    "TID", # flake8-tidy-imports
+    "UP",  # pyupgrade
+    "Q",   # flake8-quotes
+    "TCH", # flake8-type-checking
+    "T10", # flake8-debugger
+    "T20", # flake8-print
+    # TODO: "S",   # flake8-bandit
+    "YTT", # flake8-2020
+    # TODO: add more flake8 rules
+  ]
+
+  ignore = [
+    # suppression in order of # of violations:
+    "B007",
+    "T201",
+    "PLW2901",
+    "SLF001",
+    "RUF001",
+    "PLR2004",
+    "PLR0912",
+    "PLR0913",
+    "RUF005",
+    "RUF012",
+    "PLR0915",
+    "INP001",
+    "RUF015",
+    "PLR1722",
+    "ISC001",
+    "C408",
+    "B904",
+    "UP031",
+    "PLR5501",
+    "PERF203",
+    "B006",
+    "PLR1714",
+    "PERF401",
+    # TODO: these only have one violation each:
+    "SLOT000",
+    "PYI024",
+    "PLW0603",
+    "PIE800",
+    "ISC003",
+    "EXE002",
+    "C401",
+    "B028",
+    "B008",
+]
+
+
+[tool.ruff.lint.extend-per-file-ignores]
+
+"pelican/__init__.py" = [
+    # allow imports after a call to a function, see the file
+    "E402"
+]
+"**/management/commands/*" = ["T20"]
```

Comment:
What does this glob refer to?

Reply:
Good eye. That is cut-and-paste from a Django project. Removed.